src/Plugin/Field/FieldFormatter/GroupSubscribeFormatter.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Drupal\og\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Routing\RedirectDestinationTrait;
+use Drupal\Core\Url;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\EntityOwnerInterface;
+
+/**
+ * Plugin implementation for the OG subscribe formatter.
+ *
+ * @FieldFormatter(
+ *   id = "og_group_subscribe",
+ *   label = @Translation("OG Group subscribe"),
+ *   description = @Translation("Display OG Group subscribe and un-subscribe links."),
+ *   field_types = {
+ *     "og_group"
+ *   }
+ * )
+ */
+class GroupSubscribeFormatter extends FormatterBase {
+
+  use RedirectDestinationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    // We cannot use the field cache, as the formatter changes according to the
+    // user. Furthermore, this is not an expensive check, so we remove the cache
+    // entirely.
+    $elements['#cache']['max-age'] = 0;
+
+    $group = $items->getEntity();
+    $entity_type_id = $group->getEntityTypeId();
+
+    $user = User::load(\Drupal::currentUser()->id());
```

Comment:
If you let the class implement `ContainerFactoryPluginInterface` you can provide a constructor and a `::create()` method to inject the `current_user` service. Then it is not needed to make a call to `\Drupal` and break encapsulation.

Check `AddressFieldFormatter` for an example of a field formatter that uses this pattern.